cf-serverd/server_classic.c

```diff
@@ -1083,7 +1083,7 @@ int BusyWithClassicConnection(EvalContext *ctx, ServerConnectionState *conn)
     case PROTOCOL_COMMAND_AUTH_PLAIN:
         SetConnectionData(conn, (char *) (recvbuffer + strlen("CAUTH ")));
 
-        if (conn->username == NULL || !IsUserNameValid(conn->username))
```

Comment:
This is not bad code, just overcautious code. Now the compiler will optimise this away. But if in the future `username` becomes a pointer from an array, this check will be needed.